core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -198,9 +198,23 @@ public boolean isReady() {
   /**
    * Enables per-message compression, if an encoding type has been negotiated.  If no message
    * encoding has been negotiated, this is a no-op.
+   *
+   * @param enabled whether or not subsequent calls to {@link #sendMessage} should be compressed.
    */
   @ExperimentalApi
   public void setMessageCompression(boolean enabled) {
-    // noop
+    throw new UnsupportedOperationException("unimplemented");
```

Comment:
No longer noop? It seems that this method and `overrideCompressionNegotiator` is advisory, so if it was ignored (because compression was not supported) that would be fine.